### DISCUSSION TITLE
v5.0.0-rc1

### DISCUSSION THREAD

**Discussion by c-bata** [optuna/optuna#6787] [Announcements]
This is the release note of [v5.0.0-rc1](https://github.com/optuna/optuna/milestone/75?closed=1).

# Installation

Since this is a pre-release, do not forget to specify the version number to upgrade Optuna.

```
pip install optuna==5.0.0rc1
```

# Highlights

## Massive default sampler algorithm enhancement

<img width="1198" height="558" alt="Screenshot 2026-08-03 14 18 27" src="https://github.com/user-attachments/assets/e563b7e2-5f01-4d1d-9332-b46e34ccf25e" />

[Multivariate TPE](https://tech.preferred.jp/en/blog/multivariate-tpe-makes-optuna-even-more-powerful/) with a constant liar strategy and an enhanced bandwidth computation ([Watanabe 2023](https://arxiv.org/abs/2304.11127)) has become the default algorithm for single-objective optimization. Multi-Objective TPE has been adopted as the new default sampler for multi-objective optimization, replacing NSGA-II. We conducted comprehensive benchmarking, carefully selected default options, and modified implementation details to maximize optimization performance.

## Conditional PED-ANOVA (KDD 2026)

Our paper, [Conditional PED-ANOVA: Hyperparameter Importance in Hierarchical & Dynamic Search Spaces](https://arxiv.org/abs/2601.20800), has been accepted to the 32nd ACM SIGKDD Conference on Knowledge Discovery and Data Mining (KDD 2026)!

Optuna v5.0 now supports this feature and adopts it as the default algorithm for hyperparameter importance.

```python
import optuna

# Starting in Optuna v5.0, the default hyperparameter importance algorithm
# supports conditional search spaces like the one below:
def objective(trial):
    classifier_name = trial.suggest_categorical("classifier", ["SVC", "RandomForest"])
    if classifier_name == "SVC":
        svc_c = trial.suggest_float("svc_c", 1e-10, 1e10, log=True)
        classifier_obj = sklearn.svm.SVC(C=svc_c, gamma="auto")
    else:
        rf_max_depth = trial.suggest_int("rf_max_depth", 2, 32, log=True)
        classifier_obj = sklearn.ensemble.RandomForestClassifier(
            max_depth=rf_max_depth, n_estimators=10
        )
    return …

study = optuna.create_study()
study.optimize(objective)

optuna.importance.get_param_importances(study)
```

## New APIs for Constrained Optimization

Starting in Optuna v5.0, the interface for constrained optimization has changed:

- `trial.set_constraint()` is added to set constraint values.
- `trial.constraints` is added to get constraints.

Along with these new apis, `constraints_func` argument on samplers is now deprecated.

```python
import optuna

def objective(trial):
    trial.set_constraint(“c0”, c0)
    trial.set_constraint(“c1”, c1)
    …

# The constraints_func argument is now deprecated.
sampler = optuna.samplers.TPESampler()
study = optuna.create_study(sampler=samplera)
study.optimize(objective)
print(study.best_trial.constraints)
```

# Breaking Changes

- Remove `optuna.multi_objective` module (#6686)
- Remove deprecated integration wrappers for AllenNLP, Chainer, and MXNet (#6693)
- Stabilize GPSampler (#6715)
- Enable `constant_liar` by default (#6738)
- Support categorical parameters in QMCSampler (#6742, thanks @saivedant169!)
- Enable multivariate by default in TPESampler (#6746)
- Fix: remove deprecated positional-arg compatibility from study and trial APIs (#6747, thanks @yen-0!)
- Make `PedAnovaImportanceEvaluator` the default importance evaluator (#6748)
- Deprecate `constraints_func` (#6773)
- Change to treat cases without constraints as feasible (#6774)
- Normalize trial timestamps to UTC in RDBStorage and JournalStorage (#6776)
- Remove `axis_order` argument from `plot_pareto_front` (#6781)
- Remove `system_attrs` from `StudySummary` (#6782)
- Remove `categorical_distance_func` from `TPESampler` (#6783)

# New Features

- Implement qLogEI acquisition function (#6640)
- Add support for conditional search spaces in PED-ANOVA (#6682)
- Add conditional gp regressor (#6721)
- Add mutation to ga (#6724, thanks @hrntsm!)
- Use MOTPE-like split in PED-ANOVA for multi-objective studies when `target` is `None` (#6728)
- Add `constraints` property to `Trial` (#6736)
- Add `set_constraint` method to `Trial` (#6754)

# Enhancements

- Fix QMCSampler fallback to independent sampling in distributed setups (#6638, thanks @Rishabh-git10!)
- Speed up `BruteForceSampler` by avoiding full tree build based on tree size check (#6646)
- Speed up `BruteForceSampler` by candidates caching (#6650)
- Split `BruteForceSampler` refactoring [3/3] (#6657)
- Refactor PED-ANOVA (#6681)
- Add lazy tree node to `BruteForceSampler` for speedup (#6705)
- Raise `ValueError` in `PedAnovaImportanceEvaluator` for multi-objective studies without `target` (#6716)
- Fix for the GPSampler OMP issue (#6753)
- Make `TPESampler` the default sampler for multi-objective optimization (#6766)

# Bug Fixes

- Handle insufficient trials without raising `ValueError` in importances (#6720)
- Fix a concurrency issue when setting study attributes (#6751)

# Documentation

- Add ablation study tutorial using `BruteForceSampler` (#6652)
- Simplify the document for deprecated CmaEsSampler options (#6694)
- Move deprecated TPESampler options to the end (#6696)
- Fix typos in docstrings and duplicated author names in SPXCrossover citation (#6698, thanks @Divyansh-ag14!)
- Fix wording in specify params tutorial (#6711, thanks @Ryo2611!)
- Add a documentation about constrained `TPESampler` (#6712)
- Enhance `AutoSampler` citation path (#6714)
- Update sampler table (#6717)
- docs: clarify `n_warmup_steps` boundary in `MedianPruner` and `PercentilePruner` (#6733, thanks @vin0san!)
- Fix trial report doc (#6735)
- chore: consolidate instruction for external dependency (#6749)
- Update sampler table for constrained optimization (#6775)

# Tests

- Refactoring importance tests (#6725)
- Add importance evaluator test cases to optuna.testing (#6765)

# Code Fixes

- Refactor batched distribution classes to reduce branch duplication (#6689)
- Reduce redundancy in `test_brute_force.py` (#6706)
- Replace SciPy with Torch in `gp.py` (#6710)
- Fix and refactor `params` validation in PED-ANOVA (#6729)
- Show all study names upon load/create study failure (#6750)
- Changed type of study's direction into Literal | StudyDirection (#6762, thanks @yen-0!)
- Fix unused mypy ignores in `optuna/visualization/matplotlib/_rank.py` (#6767)
- Avoid unnecessary trial values in storage tests (#6769)
- Follow-up #6762: Fix mypy errors (#6778)

# Continuous Integration

- Specify workflow permissions explicitly (#6687)
- Capture Windows stdout (#6699)
- Fix logging tests broken by pytest 9.1.0 caplog behavior change (#6719)
- Hotfix for Sphinx CI (#6726)
- Fix mypy errors with latest NumPy stubs (#6730)
- Skip flaky gRPC journal storage tests (#6770)

# Other

- Bump up to version number v5.0.0.dev (#6685)
- Add `attestations: false` to fix release workflow (#6690)
- Update README with Optuna 4.9.0 release news (#6701)
- Add CODEOWNERS file (#6718)
- Bump the version up to `5.0.0rc1` (#6784)

# Thanks to All the Contributors!

This release was made possible by the authors and the people who participated in the reviews and discussions.

@Alnusjaponica, @Divyansh-ag14, @Rishabh-git10, @Ryo2611, @c-bata, @gen740, @himkt, @hrntsm, @kAIto47802, @nabenabe0928, @not522, @porink0424, @saivedant169, @sawa3030, @vin0san, @y0z, @yen-0

<hr /><em>This discussion was created from the release <a href='https://github.com/optuna/optuna/releases/tag/v5.0.0-rc1'>v5.0.0-rc1</a>.</em>
